hive table which contains binary type unavailable througn hive connector

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. create hive table:
```
create table hive_hdfs_csv_default(
col_tinyint tinyint comment "column tinyint"
,col_smallint smallint comment "column smallint"
,col_int int comment "column int"
,col_integer integer comment "column integer"
,col_bigint bigint comment "column bigint"
,col_float float comment "column float"
,col_double double comment "column double"
,col_double_precision double precision comment "column double precision"
,col_decimal decimal comment "column decimal"
,col_timestamp timestamp comment "column timestamp"
,col_date date comment "column date"
,col_string string comment "column string"
,col_varchar varchar(100) comment "column varchar"
,col_binary binary comment "column binary"
,col_char char(100) comment "column char"
,col_boolean boolean comment "column boolean"
,col_array array<int> comment "column array"
)
clustered by(col_int) sorted by(col_int) into 32 buckets
stored as textfile;
insert into hive_hdfs_csv_default select 1,1,1,1,1,1.001,10.1,110.1,1110.1,'2021-10-30 12:10:23','2021-10-30','hello world','hi hello','1110001010101011001001','nihao',true,array(1,10,100);
insert into hive_hdfs_csv_default select 2,2,2,2,2,2.001,20.1,210.1,2110.2,'2021-02-01 12:10:23','2021-02-01','hello world','hi hello','1110001010101011001001','nihao',true,array(2,20,200);
insert into hive_hdfs_csv_default select 3,3,3,3,3,3.001,30.1,310.1,3110.1,'2021-03-01 12:10:23','2021-03-01','hello world','hi abc','1110001010101011001001','nihao',true,array(3,30,300);
```

2. create catalog and select hive table:
```
create external catalog hive_catalog_1656512586190 PROPERTIES("type" = "hive","hive.metastore.uris" = "thrift://***.***.***.***:9083");
mysql> select col_int from hive_catalog_1656512586190.hive_extbl_test.hive_hdfs_csv_default limit 1;
ERROR 1064 (HY000): Unknown table 'hive_hdfs_csv_default'
```

3. fe.log:
```
2022-06-29 22:29:55,646 ERROR (starrocks-mysql-nio-pool-14|387) [HiveMetaCache.getTable():315] Failed to get table [databaseName: hive_extbl_test, tableName: hive_hdfs_csv]
java.util.concurrent.ExecutionException: com.starrocks.common.DdlException: hive table column type [BINARY] transform failed.
        at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:566) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:527) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:104) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:240) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2313) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2279) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache.get(LocalCache.java:3962) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3985) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4946) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.external.hive.HiveMetaCache.getTable(HiveMetaCache.java:313) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.hive.HiveMetadata.getTable(HiveMetadata.java:61) ~[starrocks-fe.jar:?]
        at com.starrocks.server.MetadataMgr.lambda$getTable$1(MetadataMgr.java:114) ~[starrocks-fe.jar:?]
        at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_262]
        at com.starrocks.server.MetadataMgr.getTable(MetadataMgr.java:114) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer.resolveTable(QueryAnalyzer.java:669) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer.access$100(QueryAnalyzer.java:66) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.resolveTableRef(QueryAnalyzer.java:235) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:161) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:83) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.SelectRelation.accept(SelectRelation.java:206) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:88) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryRelation(QueryAnalyzer.java:106) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:100) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:83) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:36) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:88) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer.analyze(QueryAnalyzer.java:76) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.InsertAnalyzer.analyze(InsertAnalyzer.java:34) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitInsertStatement(Analyzer.java:139) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitInsertStatement(Analyzer.java:51) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.InsertStmt.accept(InsertStmt.java:939) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:82) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:53) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:48) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:43) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:335) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:277) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:395) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:631) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_262]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_262]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_262]
Caused by: com.starrocks.common.DdlException: hive table column type [BINARY] transform failed.
        at com.starrocks.external.HiveMetaStoreTableUtils.convertColumnType(HiveMetaStoreTableUtils.java:219) ~[starrocks-fe.jar:?]
        at com.starrocks.external.HiveMetaStoreTableUtils.convertToSRTable(HiveMetaStoreTableUtils.java:241) ~[starrocks-fe.jar:?]
        at com.starrocks.external.hive.HiveMetaCache.loadTable(HiveMetaCache.java:323) ~[starrocks-fe.jar:?]
        at com.starrocks.external.hive.HiveMetaCache.access$700(HiveMetaCache.java:39) ~[starrocks-fe.jar:?]
        at com.starrocks.external.hive.HiveMetaCache$8.load(HiveMetaCache.java:144) ~[starrocks-fe.jar:?]
        at com.starrocks.external.hive.HiveMetaCache$8.load(HiveMetaCache.java:141) ~[starrocks-fe.jar:?]
        at com.google.common.cache.CacheLoader$1.load(CacheLoader.java:192) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529) ~[spark-dpp-1.0.0.jar:?]
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278) ~[spark-dpp-1.0.0.jar:?]
        ... 39 more
```

### Expected behavior (Required)
select query return the right result

### Real behavior (Required)
query error

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 52bfe63 |
+----------------------+
1 row in set (0.01 sec)
```

--- a/fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java
+++ b/fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java
@@ -150,6 +150,7 @@ public class HiveMetaStoreTableUtils {
                 return primitiveType == PrimitiveType.DATE;
             case "STRING":
             case "VARCHAR":
+            case "BINARY":
                 return primitiveType == PrimitiveType.VARCHAR;
             case "CHAR":
                 return primitiveType == PrimitiveType.CHAR ||
@@ -202,6 +203,7 @@ public class HiveMetaStoreTableUtils {
                 primitiveType = PrimitiveType.DATE;
                 break;
             case "STRING":
+            case "BINARY":
                 return ScalarType.createDefaultString();
             case "VARCHAR":
                 return ScalarType.createVarcharType(Utils.getVarcharLength(hiveType));

